packages/elastic-safer-lodash-set/lodash/_baseSet.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is forked from the lodash project (https://lodash.com/),
+ * and may include modifications made by Elasticsearch B.V.
+ * Elasticsearch B.V. licenses this file to you under the MIT License.
+ * See `packages/elastic-safer-lodash-set/LICENSE` more information.
+ */
+
+/* eslint-disable */
+
+var assignValue = require('lodash/_assignValue'),
+    castPath = require('lodash/_castPath'),
+    isIndex = require('lodash/_isIndex'),
+    isObject = require('lodash/isObject'),
+    toKey = require('lodash/_toKey');
+
+/**
+ * The base implementation of `_.set`.
+ *
+ * @private
+ * @param {Object} object The object to modify.
+ * @param {Array|string} path The path of the property to set.
+ * @param {*} value The value to set.
+ * @param {Function} [customizer] The function to customize path creation.
+ * @returns {Object} Returns `object`.
+ */
+function baseSet(object, path, value, customizer) {
+  if (!isObject(object)) {
+    return object;
+  }
+  path = castPath(path, object);
+
+  var index = -1,
+      length = path.length,
+      lastIndex = length - 1,
+      nested = object;
+
+  while (nested != null && ++index < length) {
+    var key = toKey(path[index]),
+        newValue = value;
+
+    if (typeof nested == 'function' && key == 'prototype' && hasOwnProperty.call(nested, 'prototype')) {
```

Comment:
__question:__ any reason we want to use `hasOwnProperty.call(nested, 'prototype')` in this condition? If it's just for arrow functions then behavior would be inconsistent anyway: we'd allow to set `prototype` on the first call, but fail on the second exactly same call. Or you have another use case in mind?

Reply:
I was just trying to limit the scope of when the error could be thrown to _only_ include the dangerous cases. But I didn't consider the case where you'd do it twice on the same arrow function. That would result in some weird behaviour, so I agree that we should probably not allow that. I'll update the code to not allow arrow functions either.

Reply:
Fixed 👍